Contacts

Max2002 opened the pull request.

Max2002 committed:
Made page contact, display contacts in tabular and block format.
Made statistic section.
Also made mobile version.

GCoder-13 commented on package.json:
```diff
@@ -7,14 +7,16 @@
     "prop-types": "^15.8.1",
     "react": "^18.2.0",
     "react-bootstrap": "^2.5.0",
+    "react-copy-to-clipboard": "^5.1.0",
```
delete

GCoder-13 commented on src/App.js:
```diff
@@ -4,7 +4,7 @@ import { useEffect } from 'react';
 import Header from './sections/Header';
 import Home from './sections/Home';
-import Profile from './sections/Profile/indxe';
+import Profile from './sections/Profile/index';
```
???

GCoder-13 commented on src/redux/actionCreator/getContacts.js:
```diff
@@ -0,0 +1,38 @@
+      data: { results },
+    } = await apiUser.get('/', {
+      params: {
+        seed: localStorage.getItem('auth'),
```
from redux store

GCoder-13 commented on src/sections/404/index.js:
```diff
@@ -9,14 +9,14 @@ export default function Page404() {
+    <main className={st.page404}>
       <div className={st.code}>404</div>
       <div className={st.message}>Requested page not found!</div>
       <div>
```
remove

GCoder-13 commented on src/sections/404/index.js:
```diff
@@ -9,14 +9,14 @@ export default function Page404() {
       <div className={st.code}>404</div>
       <div className={st.message}>Requested page not found!</div>
       <div>
         <Button type="button" className={st.back} onClick={backHome}>
```
Link

GCoder-13 commented on src/sections/Contacts/Table.js:
```diff
@@ -0,0 +1,116 @@
+import st from './styles.module.scss';
+
+export default function Table({ contacts }) {
+  const [sortByName, setSortByName] = useState([true, false, false]);
```
bad implementation

GCoder-13 commented on src/sections/Contacts/Table.js:
```diff
@@ -0,0 +1,116 @@
+    const dateFormat = moment(date).format('dddd, MM/DD/yyyy, h:MM:ss A');
+
+    return (
+      <tr key={email} className={st.row}>
```
???

GCoder-13 commented on src/sections/Contacts/Table.js:
```diff
@@ -0,0 +1,116 @@
+import { SortASCSvg, SortDESCSvg } from '../../assets/icons';
+import st from './styles.module.scss';
+
+export default function Table({ contacts }) {
```
create reusable table. Example: antd

GCoder-13 commented on src/sections/Contacts/index.js:
```diff
@@ -1,3 +1,101 @@
+
 export default function Contacts() {
-  return <div>Contact</div>;
+  const [switchView, setSwitchView] = useState(false);
```
rename

GCoder-13 commented on src/sections/Contacts/index.js:
```diff
@@ -1,3 +1,101 @@
+  const contacts = useSelector(dataSelector);
+  const deviceWidth = useDeviceWidth();
+
+  const handleSwitch = () => setSwitchView(!switchView);
```
rename

GCoder-13 commented on src/sections/Contacts/index.js:
```diff
@@ -1,3 +1,101 @@
+  const reloadData = () => dispatch(getContacts(1, 10));
+
+  useEffect(() => {
+    dispatch(getContacts(1, 10));
```
duplicate

GCoder-13 commented on src/sections/Contacts/index.js:
```diff
@@ -1,3 +1,101 @@
+  const deviceWidth = useDeviceWidth();
+
+  const handleSwitch = () => setSwitchView(!switchView);
+  const reloadData = () => dispatch(getContacts(1, 10));
```
rename

GCoder-13 commented on src/sections/Contacts/index.js:
```diff
@@ -1,3 +1,101 @@
+    dispatch(getContacts(1, 10));
+  }, []);
+
+  const renderBlocksView = (contact) => {
```
component

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -78,40 +78,46 @@ export default function Header() {
-            <span>Sign In</span>
-          </Button>
-        )}
+      <div className={`container ${st.containerHeader}`}>
```
clsx

GCoder-13 commented on src/assets/styles/general.scss:
```diff
@@ -40,7 +40,7 @@ input {
 }
 
 .container {
```
create component

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -78,40 +78,46 @@ export default function Header() {
+        <Portal>
+          {isActiveModal && <SignIn notify={notify} onSignIn={handleSignIn} />}
+        </Portal>
+        <ToastContainer position="top-right" autoClose={3000} />
```
ToastContainer in main component

GCoder-13 commented on src/sections/Statistic/index.js:
```diff
@@ -0,0 +1,72 @@
+  const amountGender = { male: 0, female: 0, indeterminate: 0 };
+  const amountNat = {};
+
+  contacts.map(({ gender, nat }) => {
```
bad implementation

Max2002 committed:
Several small changes

Max2002 committed:
Added section filters on page contacts and adapted for mobile version.

Max2002 committed:
Deleted section table and added components for table and cards for contacts.
Changed counting amount nationalities and genders for statistic block.
AuthKey get from redux in action getContacts.

Max2002 committed:
Added section when filters nothing return and started develop switcher for switch pages contacts

Max2002 committed:
Merge branch 'profile' into contacts

# Conflicts:
#	src/assets/styles/general.scss
#	src/assets/styles/variables.scss
#	src/sections/Header/index.js
#	src/sections/Profile/index.js

Max2002 committed:
Changed copyElement in Contacts.js and ContactCard.
Added component container for header.

Max2002 committed:
Fix filter by fullName

Max2002 committed:
Made switcher page contacts, added component for view profile and contact.
Added redirect to selected contact and added part store for contact view.

GCoder-13 commented on src/App.js:
```diff
@@ -29,10 +31,12 @@ export default function App() {
       <Routes>
         <Route index element={<Home />} />
```
Create function for render all private and all public routes

Max2002 committed:
Merge branch 'main' into contacts

GCoder-13 commented on src/components/ContactCard/index.js:
```diff
@@ -0,0 +1,51 @@
+import st from './styles.module.scss';
+
+export default function ContactCard({ contact, contactView, index, id }) {
+  const { avatar, fullName, age, email, phone, address, nat } = contact;
```
Global logic

GCoder-13 commented on src/components/ContactCard/index.js:
```diff
@@ -0,0 +1,51 @@
+        className={st.blocksAvatar}
+        src={avatar}
+        alt={fullName}
+        onClick={() => contactView(index, id)}
```
use id only

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -1,7 +1,7 @@
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import Hint from '../Hint';
+import { Hint } from '../index';
```
Infinity import

GCoder-13 commented on src/components/Table/index.js:
```diff
@@ -0,0 +1,92 @@
+
+export default function Table({ columns, options }) {
+  const [sortByName, setSortByName] = useState(null);
+  const sortOptions =
```
Code repeated, can be improved

GCoder-13 commented on src/components/Table/index.js:
```diff
@@ -0,0 +1,92 @@
+          .sort((a, b) => b.fullName[1].localeCompare(a.fullName[1]));
+
+  const renderRows = (option) => (
+    <tr>
```
???

Max2002 replied:
key

GCoder-13 commented on src/components/Table/index.js:
```diff
@@ -0,0 +1,92 @@
+  const renderRows = (option) => (
+    <tr>
+      {Object.keys(option).map((key) => (
+        <td key={key} className={key === 'email' ? st.email : null}>
```
global logic

GCoder-13 commented on src/components/Table/index.js:
```diff
@@ -0,0 +1,92 @@
+      {Object.keys(option).map((key) => (
+        <td key={key} className={key === 'email' ? st.email : null}>
+          {Array.isArray(option[key])
+            ? `${option[key][0]}. ${option[key][1]}`
```
bad

GCoder-13 commented on src/components/Table/index.js:
```diff
@@ -0,0 +1,92 @@
+      <tbody>
+        <tr className={st.firstRow}>
+          {columns.map((column) =>
+            column === 'Full name' ? (
```
global logic

GCoder-13 commented on src/components/Table/index.js:
```diff
@@ -0,0 +1,92 @@
+                onClick={handleSortByName}
+              >
+                {column}
+                {column === 'Full name' && (
```
global logic

GCoder-13 commented on src/redux/actionCreator/getContacts.js:
```diff
@@ -0,0 +1,43 @@
+    try {
+      const {
+        myProfile: { authKey },
+      } = getState();
```
use selector after getState

GCoder-13 commented on src/sections/Contacts/Filters.js:
```diff
@@ -0,0 +1,107 @@
+export default function Filters(props) {
+  const { filterByName, filterGender, filterNat, nationalities } = props;
+  const animatedComponents = makeAnimated();
+  const refFullName = useRef();
```
bad

GCoder-13 commented on src/sections/Contacts/Switcher.js:
```diff
@@ -0,0 +1,188 @@
+} from '../../assets/icons';
+import st from './styles.module.scss';
+
+export default function Switcher(props) {
```
rename

GCoder-13 replied:
full bad implementation

Max2002 committed:
Deleted files which was for card contacts

Max2002 committed:
Create function for private and public routes.
Changed component table, made reusable.
Use selector in actionCreator for get authKey in file getContacts.js.

bezuglyyvlad commented on src/sections/Contacts/SelectorPage.js:
```diff
@@ -10,7 +10,7 @@ import {
 import st from './styles.module.scss';
 
-export default function Switcher(props) {
+export default function SelectorPage(props) {
```
Rename to TablePagination

bezuglyyvlad commented on src/redux/actionCreator/getContacts.js:
```diff
@@ -23,9 +24,7 @@ export const getContacts =
-      const {
-        myProfile: { authKey },
-      } = getState();
+      const authKey = createSelector(getState, (store) => store.authKey);
```
Use existing selector

Max2002 committed:
Changed filters form and use component TextField for input filter by name.
Improved component pagination.

Max2002 committed:
Use selector authSelector for receiving authKey in actionCreators

Max2002 committed:
Fixed warning with ul
